### ISSUE TITLE
Reviews cannot be deleted from the user profile page - new pages

### ISSUE THREAD

**Issue by ioanarusiczki** [mozilla/addons-frontend#5495]
STR:
1. Load AMO dev and go to View profile page (make sure you rated a couple of extensions/themes)
2. Try to delete a review 

Expected result:
Delete button is available and the review can be deleted.

Actual result:
The review can be deleted only using the old page.

Notes:
This issue is reproducible on AMO dev and AMO stage with FF61(Win10).

![delete on old pages only](https://user-images.githubusercontent.com/33448286/42274283-c161d9e6-7f94-11e8-8f57-ba6ceba9899e.gif)

**Comment by willdurand** [comment]
There is no delete button indeed. I don't think it was specified in the mock-ups, and because we link to the addon review page, I thought we could delete the review from that page (meaning buttons here would be too much).

**Comment by willdurand** [comment]
@muffinresearch ^  – Should we ping product or UX?

**Comment by muffinresearch** [comment]
@willdurand I'd ask @jvillalobos for input.

**Comment by jvillalobos** [comment]
The review could be hard to find on the add-on review page, at least until we fix https://github.com/mozilla/addons-frontend/issues/2771. And even after fixing that, the link could be hard to discover, so I think it would be good to allow deletion. @pwalm, thoughts?

**Comment by kumar303** [comment]
Deleting reviews will be dealt with in https://github.com/mozilla/addons-frontend/issues/3075 I'm hoping that this will also apply to the User Profile pages but we might need to merge the two UserReview objects more.

**Comment by pwalm** [comment]
Yep, we should allow for deletion of user reviews. Whether or not we should leave a trace ("User deleted their review" (like Reddit)) is up for discussion.

**Comment by kumar303** [comment]
> Whether or not we should leave a trace ("User deleted their review" (like Reddit)) is up for discussion...

I checked the backend and this is possible because the delete is not deleted from the database.

**Comment by pwalm** [comment]
Don't show deletion on the front end, but we need a way internally to know if something has been posted then deleted for looking into abuse/spam/etc issues.

**Comment by kumar303** [comment]
This will be fixed by https://github.com/mozilla/addons-frontend/issues/6104

**Comment by ioanarusiczki** [comment]
The reviews or developer replies can be deleted from the user profile page
The only problem that I've noticed is with an admin account - if the reviews or dev responses are deleted from the user profile page but the detailed review page was opened before deleting in a new tab - the review/response will be displayed on the top of the page (refresh won\t help) only after navigating away from page and returning will no longer display the review/dev response. There is an open issue here https://github.com/mozilla/addons-frontend/issues/6351 
I filed a separate issue for the "Flag" button, I think that could be removed from the user profile page since your personal reviews and dev responses cannot be flagged. https://github.com/mozilla/addons-frontend/issues/6416
Verified on AMO dev with FF62 - Win 10&Android 8.0
